packages/cosmic-swingset/test/sync-xfer.test.js

```diff
@@ -0,0 +1,51 @@
+import '@endo/init/debug.js';
+import test from 'ava';
+import { makeTextDecoder, makeTextEncoder } from '../src/helpers/sync-xfer.js';
+
+test('text decoder', t => {
+  const decoder = makeTextDecoder();
+  decoder.next();
+  const buf = new TextEncoder().encode('hello world bobob');
+  let i = 0;
+  while (i < buf.length) {
```

Comment:
This looks a little too much like the implementation. I'd rather see a fixed list of buffer contents as static data.